### ISSUE TITLE
Write composer.json before adding dependencies and after adding each dependency to not lose user's composer.json.

### ISSUE THREAD

**Issue by jonpugh** [hussainweb/drupal-composer-init#4]
I went through the wizard 3 times, and each time I got stuck in the adding dependencies section.

I would search for a package, nothing would interest me, and I would not be able to choose nothing and exit the add packages loop.

If I hit ctrl+c or enter 3 times I am kicked out, and all of my changes are lost.

If you wrote the composer.json before moving onto the dependencies loop (and then wrote the dependency to composer.json on every loop) then you would avoid the user frustration of having to go through the whole wizard again.

Otherwise, this is a great tool!  Thanks for the hard work.

**Comment by ianniscamillieri** [comment]
Same problem here.   
Once you've added a few dependencies, how do you escape from the 'Search for a package' loop?

**Comment by hussainweb** [comment]
@ianniscamillieri once you have added a few dependencies, you can just enter a blank line (press the Enter key) to break the loop. Let me know if that doesn't work for some reason.

**Comment by ianniscamillieri** [comment]
@hussainweb You're right, it works perfectly. I've tried 'no', ESCAPE, CTRL-C because I had the feeling there was something to type to say 'stop'.

I agree that a little more detailed documentation would be useful the README. Just to explain the steps and give some infos about what to answer at each prompt.

Why don't add these links at least?
- https://www.drupal.org/files/composer-drupal-init.gif
- https://www.drupal.org/files/d7-composer-init.gif

Anyway, thanks a lot for the support and congrats for your script!!!

**Comment by ianniscamillieri** [comment]
However, if you have searched for a package, and then change your mind for any reason, how do you escape from `Enter package # to add, or the complete package name if it is not listed:`?
